src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -514,6 +528,27 @@ internal NamedTypeSymbol UnderlyingTupleType
 
         internal bool HasFriendlyNames => _hasFriendlyNames;
 
+        public override ImmutableArray<ITypeSymbol> TupleElementTypes
+        {
+            get
+            {
+                return _fields.SelectAsArray(f => (ITypeSymbol)f.Type);
```

Comment:
Yes, a tuple always has types (even a theoretical 0-tuple would have an empty list of tuples), but it may have no friendly names (as in default/null). I think the current behavior is correct.